### DISCUSSION TITLE
Brown image generated on a PC with no discreet GPU

### DISCUSSION THREAD

**Discussion by AshD** [saddam213/OnnxStack#54] [Q&A]
One of my users who does not have a discreet GPU is only seeing a brown image. No errors. 
See attached.

Their PC Configuration: 
CPU: 11th Gen Intel(R) Core(TM) i5-11400 @ 2.60GHz, Speed: 2592MHz, Cores: 6
GPU: Intel(R) UHD Graphics 730, Memory: 1024 MB
Memory: Total: 32 GB, Free: 25 MB
OS: Microsoft Windows 11 Home, Version: 10.0.22621

Model used: softwareweaver/photon
Prompt: Cat

I can't repro this and was wondering if anyone else saw something similar Or have any tips to help resolve this issue.

Thanks,
Ash
![unnamed](https://github.com/saddam213/OnnxStack/assets/233016/b9d8eea5-a12f-47a7-9c00-935cab78548d)

**Comment by AshD** [comment]
I managed to repro this on a Laptop with a 3070 GPU that was generating images earlier. 

I had updated it with Windows update and it started generating brown images like the one above. 

It turned out that the Windows update made Intel UHD graphics the default GPU, so when Onnx Stable Diffusion API was called with DeviceIDd=0 , it was trying to use the UHD graphics with does not support DirectML optimized models. When I set the DeviceID=1, it worked!

One way would be to search for discreet GPU and set the correct Device Id at the app layer, the other would be to set DeviceId to -1 to find and use the discreet GPU inside OnnxStack. 

Thoughts @saddam213 and others?

Thanks,
Ash

**Reply by saddam213** [reply]
I assume getting the device list will be different for each Execution Provider and may require dragging in some 3rd party libs (DirectX for CUDA/RT devices perhaps)

However device detection is probably out of scope for OnnxStack, but would be something to look at for a more advanced UI app

Hopefully once the core library has more features someone will make a new UI or web app

**Reply by saddam213** [reply]
Managed to replicate this, its not that the device does not support DirectML its the lack of VRAM, if I fill my VRAM and overflow into shared VRAM all images come out like this, brown or grey

Since you can assign individual model files to a device you could take advantage of that device, assign the text models to that device in advanced, TASED might even squeeze on there :p

**Comment by AshD** [comment]
That makes sense. The Intel on chip GPU probably does not have much VRAM.
